crates/router/src/routes/app.rs

```diff
@@ -1150,25 +1152,35 @@ impl PaymentMethods {
     pub fn server(state: AppState) -> Scope {
         let mut route = web::scope("/v2/payment-methods").app_data(web::Data::new(state));
         route = route
-            .service(web::resource("").route(web::post().to(create_payment_method_api)))
-            .service(
-                web::resource("/create-intent")
-                    .route(web::post().to(create_payment_method_intent_api)),
-            )
-            .service(
-                web::resource("/{id}/confirm-intent")
-                    .route(web::post().to(confirm_payment_method_intent_api)),
-            )
             .service(
-                web::resource("/{id}/update-saved-payment-method")
-                    .route(web::patch().to(payment_method_update_api)),
+                web::resource("").route(web::post().to(payment_methods::create_payment_method_api)),
             )
             .service(
-                web::resource("/{id}")
-                    .route(web::get().to(payment_method_retrieve_api))
-                    .route(web::delete().to(payment_method_delete_api)),
+                web::resource("/create-intent")
+                    .route(web::post().to(payment_methods::create_payment_method_intent_api)),
             );
 
+        route = route.service(
+            web::scope("/{id}")
+                .service(
+                    web::resource("")
+                        .route(web::get().to(payment_methods::payment_method_retrieve_api))
+                        .route(web::delete().to(payment_methods::payment_method_delete_api)),
+                )
+                .service(
+                    web::resource("/list-payment-methods")
+                        .route(web::get().to(payment_methods::list_payment_methods)),
+                )
+                .service(
+                    web::resource("/confirm-intent")
+                        .route(web::post().to(payment_methods::confirm_payment_method_intent_api)),
+                )
+                .service(
+                    web::resource("/update-saved-payment-method")
+                        .route(web::patch().to(payment_methods::payment_method_update_api)),
```

Comment:
Is this supposed to be `put()`, to keep things consistent with the other v2 APIs?

Reply:
@Sarthak1799